modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/resources/Key.scala

```diff
@@ -0,0 +1,23 @@
+package ch.epfl.bluebrain.nexus.kg.core.resources
+
+import io.circe.{Decoder, Encoder, Json}
+
+/**
+  * The unique key which identifies a representation of a resource
+  *
+  * @param projectRef the identifier for the project. TODO: Change the type to a [ProjectReference]
+  * @param resourceId the unique identifier (inside this project) of the current resource. TODO: Change the type to a [IRI] or [Id]
+  * @param schemaId   the unique identifier (inside this project) of the schema which constrains this resource. TODO: Change the type to a [IRI] or [Id]
+  */
+final case class Key(projectRef: String, resourceId: String, schemaId: String)
+object Key {
+
+  implicit val encodeKey: Encoder[Key] = Encoder.encodeJson.contramap {
+    case Key(project, resource, schema) =>
+      Json.obj("@id"     -> Json.fromString(resource),
+               "project" -> Json.fromString(project),
```

Comment:
maybe we should also handle the `project` as an `@id` (in terms of jsonld) to ensure an accurate Event description in JsonLd